examples/tools/twolayeredcache

```diff
@@ -0,0 +1,185 @@
+module twolayeredcache {
+    //**********************************************************
+    // TYPE DEFINITIONS
+    //**********************************************************
+    type CacheLayer = Set[int]
+    type ClientPID = int
+
+    // tpe: "read" or "write"
+    type HistoryEntry = { tpe: str, value: int}
+
+    //**********************************************************
+    // CONSTANTS
+    //**********************************************************
+    pure val DefaultExpireDuration = 3
+    pure val ClientProcesses: Set[ClientPID] = 1.to(10)
+    pure val MaxVal = 1000000
+    pure val Expired = -99
+    pure val NotFound = -98
+    
+    //**********************************************************
+    // STATE MACHINE
+    // State-dependent definitions and actions
+    //**********************************************************
+
+    var l1: CacheLayer
+    var l2: CacheLayer
+    var num: int
+
+    // Global log of system events. Use to specify correctness
+    // properties below.
+    var history: List[HistoryEntry]
+
+    //**********************************************************
+    // FUNCTIONAL LAYER
+    // Values and functions that are state-independent
+    //**********************************************************
+
+    action writeLayer(layer: CacheLayer, v: int): bool = all {
+        layer' = layer.union(Set(v))
+    }
```

Comment:
Passing the state variable "as reference" here and assigning to a parameter is not valid Quint. I noticed Apalache is working with that, I guess because it's made for TLA+ and TLA+ allows arbitrary usage of the prime operator (`'`). In Quint, you can only use it directly with state variables, so you'd actually have to use something like this:

```suggestion
    action writeL1(v: int): bool = all {
        l1' = l1.union(Set(v))
    }

    action writeL2(v: int): bool = all {
        l2' = l2.union(Set(v))
    }
```

Using the Quint simulator (`quint run`) with the current definition results in a Runtime Error.